packages/hardhat-core/test/internal/hardhat-network/provider/HardhatBlockchain.ts

```
@@ -79,9 +79,12 @@ describe("HardhatBlockchain", () => {
       assert.equal(await blockchain.getBlock(1n), one);
     });
 
-    it("returns undefined non-existent block", async () => {
-      assert.equal(await blockchain.getBlock(0), undefined);
-      assert.equal(await blockchain.getBlock(randomHashBuffer()), undefined);
+    it("throws error for non-existent block", async () => {
+      await assert.isRejected(blockchain.getBlock(0), "Block not found");
```

Some tests changed because `getBlock` now throws instead of returning null.